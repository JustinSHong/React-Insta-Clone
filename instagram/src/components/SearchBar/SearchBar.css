.SearchBar {
	border-bottom: 0.5px solid gray;
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin: 0 1%;
	padding: 0 1%;
}

.brand {
	/*border: 1px solid red;*/
	display: inline-block;
	align-items: center;
}

.brand a, .brand p {
	display: inline-block;
}

.brand a {
	border-right: 1px solid black;
	padding-right: 20px;
}

.brand p {
	padding-left: 20px;
}

.SearchBar a {
	color: black;
}

.accentLogo {
	margin: 0 20px;
}