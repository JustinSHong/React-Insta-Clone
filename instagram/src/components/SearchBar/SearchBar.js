import React from 'react';

class SearchBar extends React.Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	render() {
		return (
			<div>
				<h1>Search Bar goes Here</h1>
			{/* brand icon goes here */}
		{/* brand title goes here */}
	{/* search bar goes here */}
{/* accent icons go here */}
			</div>
		);
	}
}

export default SearchBar;