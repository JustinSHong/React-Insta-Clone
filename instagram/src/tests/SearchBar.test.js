import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
// data
import postData from "../dummy-data";
// component
import SearchBar from "../components/SearchBar/SearchBar";

Enzyme.configure({ adapter: new Adapter() });

describe("<SearchBar />", () => {
	const search = jest.fn();
	const tree = renderer.create(<SearchBar search={search} />);
	it("should render without crashing", () => {
		expect(tree).toMatchSnapshot();
	});

	it("should render an input tag", () => {
		const component = shallow(<SearchBar search={search} />);
		expect(component.find("input").length).toBe(1);
	});

	it("should have an input that changes", () => {
		const component = shallow(<SearchBar search={search} />);
		const input = component.find("input");
		const profile = "google";
		expect(input.value).toBe(undefined);
		input.value = profile;
		expect(input.value).toEqual("google");
	});

	it("should call handleSearch to change state", () => {
		const handleSearch = jest.fn();
		const component = shallow(<SearchBar search={search} />);
		const input = component.find("input");
		const event = { target: { name: `${input.props().name}`, value: "abc" } };

		expect(component.state("user").length).toBe(0);
		component.instance().handleSearch(event);
		expect(component.state("user")).toEqual("abc");
	});

	it("should render a links to instagram.com", () => {
		const component = shallow(<SearchBar search={search} />);
		const links = component.find("a");
		links.forEach((node, index) => {
			expect(node.props().href).toBe("https://www.instagram.com/?hl=en");
		});
	});
});
