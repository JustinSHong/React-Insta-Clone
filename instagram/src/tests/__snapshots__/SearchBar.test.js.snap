// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render without crashing 1`] = `
<div
  className="SearchBar"
>
  <div
    className="brand"
  >
    <a
      href="https://www.instagram.com/?hl=en"
    >
      <i
        className="fab fa-instagram fa-3x"
      />
    </a>
    <p>
      Instagram
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="user"
      onChange={[Function]}
      placeholder="Search here..."
      type="text"
      value=""
    />
  </form>
  <div>
    <a
      className="accentLogo"
      href="https://www.instagram.com/?hl=en"
    >
      <i
        className="far fa-compass fa-2x"
      />
    </a>
    <a
      className="accentLogo"
      href="https://www.instagram.com/?hl=en"
    >
      <i
        className="far fa-heart fa-2x"
      />
    </a>
    <a
      className="accentLogo"
      href="https://www.instagram.com/?hl=en"
    >
      <i
        className="far fa-user fa-2x"
      />
    </a>
  </div>
</div>
`;
